### ISSUE TITLE
Accessibility: Action buttons in header: aria-labelledby attribute is not allowed on the generic role

### ISSUE THREAD

**Issue by msiewierska** [shopware/platform#13975]
### Shopware Version

6.7.3.1

### Affected area / extension

Platform(Default)

### Actual behaviour

WCAG 2.1 success criterion A 4.1.2 is violated with error:  aria-labelledby attribute is not allowed on the generic role

<img width="1627" height="309" alt="Image" src="https://github.com/user-attachments/assets/9dd57317-c7fd-4228-aee5-2096f200170b" />

### Expected behaviour

Because a plain <span> exposes the implicit ARIA role generic, some testing tools flag any non-global ARIA attributes (such as aria-labelledby) as invalid on it. To keep wishlist and account badges focusable and let it reference another element for its label, give it an explicit, valid interactive role (or turn it into a native interactive element) and ensure it’s keyboard reachable.

### How to reproduce

Enable wishlist. Perform accessibility check for header.
